include/deal.II/differentiation/ad/sacado_math.h

```diff
@@ -0,0 +1,96 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_differentiation_ad_sacado_math_h
+#define dealii_differentiation_ad_sacado_math_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
+
+#include <deal.II/base/numbers.h>
+#include <deal.II/differentiation/ad/ad_number_traits.h>
+#include <deal.II/differentiation/ad/sacado_number_types.h>
+
+
+#ifndef DOXYGEN
+
+/**
+ * Define some missing fundamental math functions
+ */
+namespace std
```

Comment:
maybe we should open dealii namespace first so that everything deal.II defines stays in its own namespace. 
Internally, that should not matter as you use `std::something` within the `dealii::` namespace, so i guess it shall blend ok with the actual `std::` namespace.

Reply:
Interesting idea. I'll test to see if this works as expected.

Reply:
Nope, sadly this doesn't work. It confuses the compiler completely to have a `std` namespace within the `dealii` namespace.

Reply:
pity.

Reply:
I would offer to post the compiler error but it so long I don't know if it would even make sense. It breaks the code in places that don't even touch these files!

Reply:
Yes, you can't do that because then for every function anywhere in `namespace dealii`, whenever we write `std::sqrt` it will find `namespace dealii::std` and then complain about not finding a function of name `sqrt` in it. C++ does not backtrack at that point to find other namespaces named `std`. In other words, you can't do that unless we want to qualify everything with `::std::sqrt` -- which I think would not be a good idea.